game/src/edit/roads.rs

```diff
@@ -1,84 +1,461 @@
+                            new.lanes_ltr.push(LaneSpec {
+                                lt,
+                                dir: Direction::Fwd,
+                                width: NORMAL_LANE_THICKNESS,
+                            });
+                        }));
+                    apply_map_edits(ctx, app, edits);
+                    self.redo_stack.clear();
+
+                    assert!(self.current_lane.is_none());
+                    self.current_lane =
+                        Some(app.primary.map.get_r(self.r).lanes_ltr().last().unwrap().0);
+                    self.main_panel =
+                        make_main_panel(ctx, app, app.primary.map.get_r(self.r), self.current_lane);
+                    self.highlight_selection =
+                        highlight_current_selection(ctx, app, self.r, self.current_lane);
+                    self.top_panel = make_top_panel(
+                        ctx,
+                        app,
+                        self.num_edit_cmds_originally,
+                        self.redo_stack.is_empty(),
+                    );
+                } else {
+                    unreachable!()
+                }
+            }
+            Outcome::Changed => {
+                let speed_limit = self.main_panel.dropdown_value("speed limit");
+                if speed_limit != app.primary.map.get_r(self.r).speed_limit {
+                    let mut edits = app.primary.map.get_edits().clone();
+                    edits
+                        .commands
+                        .push(app.primary.map.edit_road_cmd(self.r, |new| {
+                            new.speed_limit = speed_limit;
+                        }));
+                    apply_map_edits(ctx, app, edits);
+                    self.redo_stack.clear();
+
+                    // Lane IDs don't change
+                    self.main_panel =
+                        make_main_panel(ctx, app, app.primary.map.get_r(self.r), self.current_lane);
+                    self.top_panel = make_top_panel(
+                        ctx,
+                        app,
+                        self.num_edit_cmds_originally,
+                        self.redo_stack.is_empty(),
+                    );
+                } else {
+                    let width = self.main_panel.dropdown_value("width");
+                    self.modify_current_lane(ctx, app, Some(0), |new, idx| {
+                        new.lanes_ltr[idx].width = width;
+                    });
+                }
+            }
+            _ => {}
+        }
+
+        let mut highlight = self.current_lane;
+        if let Some(name) = self.main_panel.currently_hovering() {
+            if let Some(idx) = name.strip_prefix("modify Lane #") {
+                highlight = Some(LaneID(idx.parse().unwrap()));
+            }
+        }
+        if highlight != self.highlight_selection.0 {
+            self.highlight_selection = highlight_current_selection(ctx, app, self.r, highlight);
+        }
```

Comment:
I like how hovering the control highlights the lane, nice!

Reply:
I want to implement the inverse, because I found myself trying to click on the map, but I ran out of steam making the two work together. Will retry!

Reply:
Ah yeah, hovering or selecting using the actual lane on the map would make sense!